### ISSUE TITLE
[CI] MetadataIndexTemplateServiceTests testFindV2TemplatesForDateMathIndex failing

### ISSUE THREAD

**Issue by bpintea** [elastic/elasticsearch#87788]
`failed to process cluster event (remove-index-template-v2 [*]) within 30s` -- not positive if actual test failure or environment-induced (slow machine?).

**Build scan:**
https://gradle-enterprise.elastic.co/s/zel3uos2s5uo6/tests/:server:test/org.elasticsearch.cluster.metadata.MetadataIndexTemplateServiceTests/testFindV2TemplatesForDateMathIndex

**Reproduction line:**
`./gradlew ':server:test' --tests "org.elasticsearch.cluster.metadata.MetadataIndexTemplateServiceTests.testFindV2TemplatesForDateMathIndex" -Dtests.seed=BDB5435266EEBA60 -Dtests.locale=ar-DZ -Dtests.timezone=America/Chicago -Druntime.java=17`

**Applicable branches:**
master

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.cluster.metadata.MetadataIndexTemplateServiceTests&tests.test=testFindV2TemplatesForDateMathIndex

**Failure excerpt:**
```
org.elasticsearch.cluster.metadata.ProcessClusterEventTimeoutException: failed to process cluster event (remove-index-template-v2 [*]) within 30s

  at org.elasticsearch.cluster.service.MasterService$Batcher.lambda$onTimeout$0(MasterService.java:152)
  at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:709)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
  at java.lang.Thread.run(Thread.java:833)

```

**Comment by iverase** [comment]
similar failure in https://gradle-enterprise.elastic.co/s/r2llaatzf5zdi and https://gradle-enterprise.elastic.co/s/47gdawgxddk4s.
It seems the whole suite timed out, it does not reproduce locally.

**Comment by martijnvg** [comment]
Closed in favour of #87948
